### PR TITLE
Relocate PFW ephemeris decoding code to LSN

### PR DESCRIPTION
**Pull request by adel-mamin**
The planning issue: https://swift-nav.atlassian.net/browse/STAR-953

This PR was created to move BDS ephemeris decoding code from PFW. It is done to re-use the code from gnss-converters repository.

### CODE REVIEW

**Comment by pgrgich** [comment]
This is all RTCM decoding, right? If so, then it should go in `librtcm`, I think. https://github.com/swift-nav/librtcm/pull/93/ is well on the way to being merged, so we should collaborate on that.

**Comment by adel-mamin** [comment]
> This is all RTCM decoding, right? If so, then it should go in librtcm, I think. swift-nav/librtcm#93 is well on the way to being merged, so we should collaborate on that.

This PR was added to support https://github.com/swift-nav/gnss-converters/pull/319. The decoders in the PR expect input data in the form of raw broadcast data and produce the output in the form of `ephemeris_t`. https://github.com/swift-nav/gnss-converters/pull/319 then takes `ephemeris_t` and encodes it as SBP. So it is not RTCM decoding.
As mentioned in https://swift-nav.atlassian.net/browse/STAR-953 the idea was to relocate broadcast navigation data decoding from piksi_firmware_private repo to libswiftnav and then re-use the decoders from gnss-converters repo.

File: src/ephemeris.c [hunk 1]
```diff
@@ -41,6 +41,27 @@
 
 u32 decode_fit_interval(u8 fit_interval_flag, u16 iodc);
 
+/**
+ * Beidou URA table.
+ * Reference: BDS-SIS-ICD-2.1
+ */
+static const float bds_ura_table[16] = {[0] = 2.0f,
```
Comment (pgrgich) [inline src/ephemeris.c]: Can we unify this with https://github.com/swift-nav/gnss-converters/blob/master/c/src/rtcm3_sbp_ephemeris.c#L63?

File: include/swiftnav/ephemeris.h [hunk 2]
```diff
@@ -34,6 +34,10 @@ extern "C" {
 #define EPH_SOURCE_GAL_INAV 0
 #define EPH_SOURCE_GAL_FNAV 1
 
+#ifndef BDS_FIT_INTERVAL_SECONDS
```
Comment (jbangelo) [inline include/swiftnav/ephemeris.h]: Where else would `BDS_FIT_INTERVAL_SECONDS` be defined? Conditionally defining preprocessor items like this seems like a code smell

Comment (adel-mamin) [reply include/swiftnav/ephemeris.h]: Here: https://github.com/swift-nav/piksi_firmware_private/blob/master/src/nav_msg/nav_msg_bds.h#L49-L55
Once this PR is merged the BDS ephemeris decoding code with the define should be removed from PFW repo. Then this `ifndef` clause could be removed as well.

File: src/ephemeris.c (hunk 1 above)
Comment (adel-mamin) [reply src/ephemeris.c]: Done by https://github.com/swift-nav/libswiftnav/pull/119/commits/6553a82a53f73ae9ed47bd16967785a19d0694e6 and https://github.com/swift-nav/gnss-converters/pull/319/commits/448d90f85200f00ece962c94a98380e980b593a2

**Pull request merged**